### ISSUE TITLE
[MEAR-201] Exception during parallel execution (

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-ear-plugin#377]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** opened **[MEAR-201](https://issues.apache.org/jira/browse/MEAR-201?redirect=false)** and commented

```
00:15:15.602 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear (default-ear) on project xyz-ear: Execution default-ear of goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear failed:  file mode must be 3 or 4 characters -> [Help 1]
00:15:15.602 org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear (default-ear) on project xyz-ear: Execution default-ear of goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear failed:  file mode must be 3 or 4 characters
00:15:15.602 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
00:15:15.602 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
00:15:15.602 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
00:15:15.602 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
00:15:15.602 	at org.apache.maven.lifecycle.internal.LifecycleThreadedBuilder$1.call(LifecycleThreadedBuilder.java:169)
00:15:15.602 	at org.apache.maven.lifecycle.internal.LifecycleThreadedBuilder$1.call(LifecycleThreadedBuilder.java:165)
00:15:15.602 	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:314)
00:15:15.602 	at java.util.concurrent.FutureTask.run(FutureTask.java:149)
00:15:15.602 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:452)
00:15:15.602 	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:314)
00:15:15.602 	at java.util.concurrent.FutureTask.run(FutureTask.java:149)
00:15:15.602 	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:897)
00:15:15.602 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:919)
00:15:15.602 	at java.lang.Thread.run(Thread.java:770)
00:15:15.602 Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-ear of goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear failed:  file mode must be 3 or 4 characters
00:15:15.602 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:115)
00:15:15.602 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
00:15:15.602 	... 13 more
00:15:15.602 Caused by: java.lang.IllegalArgumentException:  file mode must be 3 or 4 characters
00:15:15.602 	at org.codehaus.plexus.archiver.util.FilePermissionUtils.getFilePermissionFromMode(FilePermissionUtils.java:55)
00:15:15.602 	at org.codehaus.plexus.archiver.util.ArchiveEntryUtils.applyPermissionsWithJvm(ArchiveEntryUtils.java:133)
00:15:15.602 	at org.codehaus.plexus.archiver.util.ArchiveEntryUtils.chmod(ArchiveEntryUtils.java:61)
00:15:15.602 	at org.codehaus.plexus.archiver.zip.AbstractZipUnArchiver.extractFile(AbstractZipUnArchiver.java:238)
00:15:15.602 	at org.codehaus.plexus.archiver.zip.AbstractZipUnArchiver.extractFileIfIncluded(AbstractZipUnArchiver.java:185)
00:15:15.602 	at org.codehaus.plexus.archiver.zip.AbstractZipUnArchiver.execute(AbstractZipUnArchiver.java:149)
00:15:15.602 	at org.codehaus.plexus.archiver.AbstractUnArchiver.extract(AbstractUnArchiver.java:120)
00:15:15.602 	at org.apache.maven.plugin.ear.EarMojo.changeManifestClasspath(EarMojo.java:669)
00:15:15.602 	at org.apache.maven.plugin.ear.EarMojo.execute(EarMojo.java:338)
00:15:15.602 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:106)
00:15:15.602 	... 14 more
00:15:15.603 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear (default-ear) on project abc-ear: Execution default-ear of goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear failed:  file mode must be 3 or 4 characters -> [Help 1]
00:15:15.603 org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear (default-ear) on project abc-ear: Execution default-ear of goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear failed:  file mode must be 3 or 4 characters
00:15:15.603 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
00:15:15.603 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
00:15:15.603 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
00:15:15.603 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
00:15:15.603 	at org.apache.maven.lifecycle.internal.LifecycleThreadedBuilder$1.call(LifecycleThreadedBuilder.java:169)
00:15:15.603 	at org.apache.maven.lifecycle.internal.LifecycleThreadedBuilder$1.call(LifecycleThreadedBuilder.java:165)
00:15:15.603 	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:314)
00:15:15.603 	at java.util.concurrent.FutureTask.run(FutureTask.java:149)
00:15:15.603 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:452)
00:15:15.603 	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:314)
00:15:15.603 	at java.util.concurrent.FutureTask.run(FutureTask.java:149)
00:15:15.603 	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:897)
00:15:15.603 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:919)
00:15:15.603 	at java.lang.Thread.run(Thread.java:770)
00:15:15.603 Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-ear of goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear failed:  file mode must be 3 or 4 characters
00:15:15.603 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:115)
00:15:15.603 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
00:15:15.603 	... 13 more
00:15:15.603 Caused by: java.lang.IllegalArgumentException:  file mode must be 3 or 4 characters
00:15:15.603 	at org.codehaus.plexus.archiver.util.FilePermissionUtils.getFilePermissionFromMode(FilePermissionUtils.java:55)
00:15:15.603 	at org.codehaus.plexus.archiver.util.ArchiveEntryUtils.applyPermissionsWithJvm(ArchiveEntryUtils.java:133)
00:15:15.603 	at org.codehaus.plexus.archiver.util.ArchiveEntryUtils.chmod(ArchiveEntryUtils.java:61)
00:15:15.603 	at org.codehaus.plexus.archiver.zip.AbstractZipUnArchiver.extractFile(AbstractZipUnArchiver.java:238)
00:15:15.603 	at org.codehaus.plexus.archiver.zip.AbstractZipUnArchiver.extractFileIfIncluded(AbstractZipUnArchiver.java:185)
00:15:15.603 	at org.codehaus.plexus.archiver.zip.AbstractZipUnArchiver.execute(AbstractZipUnArchiver.java:149)
00:15:15.603 	at org.codehaus.plexus.archiver.AbstractUnArchiver.extract(AbstractUnArchiver.java:120)
00:15:15.603 	at org.apache.maven.plugin.ear.EarMojo.changeManifestClasspath(EarMojo.java:669)
00:15:15.603 	at org.apache.maven.plugin.ear.EarMojo.execute(EarMojo.java:338)
00:15:15.603 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:106)
00:15:15.603 	... 14 more
00:15:15.603 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear (default-ear) on project middletier-ear: Execution default-ear of goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear failed:  file mode must be 3 or 4 characters -> [Help 1]
00:15:15.603 org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear (default-ear) on project middletier-ear: Execution default-ear of goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear failed:  file mode must be 3 or 4 characters
00:15:15.603 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
00:15:15.603 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
00:15:15.603 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
00:15:15.603 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
00:15:15.603 	at org.apache.maven.lifecycle.internal.LifecycleThreadedBuilder$1.call(LifecycleThreadedBuilder.java:169)
00:15:15.603 	at org.apache.maven.lifecycle.internal.LifecycleThreadedBuilder$1.call(LifecycleThreadedBuilder.java:165)
00:15:15.603 	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:314)
00:15:15.603 	at java.util.concurrent.FutureTask.run(FutureTask.java:149)
00:15:15.603 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:452)
00:15:15.603 	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:314)
00:15:15.604 	at java.util.concurrent.FutureTask.run(FutureTask.java:149)
00:15:15.604 	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:897)
00:15:15.604 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:919)
00:15:15.604 	at java.lang.Thread.run(Thread.java:770)
00:15:15.604 Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-ear of goal org.apache.maven.plugins:maven-ear-plugin:2.9.1:ear failed:  file mode must be 3 or 4 characters
00:15:15.604 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:115)
00:15:15.604 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
00:15:15.604 	... 13 more
00:15:15.604 Caused by: java.lang.IllegalArgumentException:  file mode must be 3 or 4 characters
00:15:15.604 	at org.codehaus.plexus.archiver.util.FilePermissionUtils.getFilePermissionFromMode(FilePermissionUtils.java:55)
00:15:15.604 	at org.codehaus.plexus.archiver.util.ArchiveEntryUtils.applyPermissionsWithJvm(ArchiveEntryUtils.java:133)
00:15:15.604 	at org.codehaus.plexus.archiver.util.ArchiveEntryUtils.chmod(ArchiveEntryUtils.java:61)
00:15:15.604 	at org.codehaus.plexus.archiver.zip.AbstractZipUnArchiver.extractFile(AbstractZipUnArchiver.java:238)
00:15:15.604 	at org.codehaus.plexus.archiver.zip.AbstractZipUnArchiver.extractFileIfIncluded(AbstractZipUnArchiver.java:185)
00:15:15.604 	at org.codehaus.plexus.archiver.zip.AbstractZipUnArchiver.execute(AbstractZipUnArchiver.java:149)
00:15:15.604 	at org.codehaus.plexus.archiver.AbstractUnArchiver.extract(AbstractUnArchiver.java:120)
00:15:15.604 	at org.apache.maven.plugin.ear.EarMojo.changeManifestClasspath(EarMojo.java:669)
00:15:15.604 	at org.apache.maven.plugin.ear.EarMojo.execute(EarMojo.java:338)
00:15:15.604 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:106)
00:15:15.604 	... 14 more
00:15:15.604 [ERROR] 
00:15:15.604 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
00:15:15.604 [ERROR] 
00:15:15.604 [ERROR] For more information about the errors and possible solutions, please read the following articles:
00:15:15.604 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
00:15:15.604 [ERROR] 
00:15:15.604 [ERROR] After correcting the problems, you can resume the build with the command
00:15:15.605 [ERROR]   mvn <goals> -rf :xyz-ear
```

---
No further details from [MEAR-201](https://issues.apache.org/jira/browse/MEAR-201?redirect=false)

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Hi Kristian,
it's not that easy, cause it's at customer where i maintain a large build and was trying to use parallized build which caused the above...after diving into this i can reproduce it now a little bit better. The exception has occured while using `maven-war-plugin:2.5` which is used to create war files where the maven-ear-plugin is using skinnyWars which means they will be unpacked during the creation of the ear file. After changing back to `maven-war-plugin:2.4` it works without any issue. So i would assume it is more located in maven-war-plugin based on the upgrade. I will try and get more informations about this...give me some time...

**Comment by jira-importer** [comment]
**[Kristian Rosenvold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=krosenvold)** commented

You may want to test the PLXCOMP-258 fix on this one ø

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

So works now.

**Comment by jira-importer** [comment]
**[Karl Heinz Marbaise](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=khmarbaise)** commented

Has been fixed with Release 2.10.
